### ISSUE TITLE
Экспортный метод может находится только в областях ПрограммногоИнтерфейса

### ISSUE THREAD

**Issue by asosnoviy** [1c-syntax/bsl-language-server#69]

**Comment by zeegin** [comment]
Неверно: В ОписанииОповещения можно вызывать только экспортные методы. Потому в клиентском коде очень много экспортных служебных методов. Помимо этого есть некоторые модули служебные и с вызовом сервера которые не содержат программного инетерфейса но содержат механизм внутреннего проброса значений с клиента на сервер в рамках подсистемы.

Для того чтобы не смущать потребителей мы придумали такую штуку:
В модуле, где находится программный интерфейс не должно быть экспортных методов кроме программного интерфейса. В модулях, где нет программного интерфейса (обычно это модули с суффиксом Служебный (англ. Internal)) могут быть служебные экспортные.

Такой подход позволяет при наборе методов по точке не видеть всю служебную фигню, которая потребителям неинтересна.

**Comment by asosnoviy** [comment]
Вот то ли стандарт изменился, то ли я его раньше неправильно читал.

* Раздел «Программный интерфейс» содержит экспортные процедуры и функции, предназначенные для использования другими объектами конфигурации или другими программами (например, через внешнее соединение).
* Раздел «Служебный программный интерфейс»  предназначен для модулей, которые являются частью некоторой функциональной подсистемы. В нем должны быть размещены экспортные процедуры и функции, которые допустимо вызывать только из других функциональных подсистем этой же библиотеки.
* Раздел «Служебные процедуры и функции» содержит процедуры и функции, составляющие внутреннюю реализацию общего модуля. В тех случаях, когда общий модуль является частью некоторой функциональной подсистемы, включающей в себя несколько объектов метаданных, в этом разделе также могут быть размещены служебные экспортные процедуры и функции, предназначенные только для вызова из других объектов данной подсистемы. 

Соответственно без анализа метаданных по подсистемам(Библиотекам). По честному реализовать проверку нельзя.
